### PR TITLE
Added support for status code list in HTTP Chaos

### PR DESCRIPTION
**Pull request by avaakash**
Signed-off-by: Akash Shrivastava <as86414@gmail.com>

**What this PR does / why we need it**:
This PR adds support for providing multiple status code in the HTTP Status Code Chaos experiment. A random status code from the provided values will be selected and used. 
**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Checklist:**
-   [ ] Fixes #<issue number>
-   [ ] PR messages has document related information
-   [ ] Labelled this PR & related issue with `breaking-changes` tag
-   [ ] PR messages has breaking changes related information
-   [ ] Labelled this PR & related issue with `requires-upgrade` tag
-   [ ] PR messages has upgrade related information
-   [ ] Commit has unit tests
-   [ ] Commit has integration tests
-   [ ] E2E run Required for the changes

### CODE REVIEW

File: chaoslib/litmus/http-chaos/lib/statuscode/status-code.go [hunk 1]
```diff
@@ -29,27 +35,49 @@ func PodHttpStatusCodeChaos(experimentsDetails *experimentTypes.ExperimentDetail
...
 // 1. It checks if the status code is provided or not. If it's not then it selects a random status code from supported list
 // 2. It checks if the provided status code is valid or not.
-func GetStatusCode(statusCode int) (int, error) {
-	if statusCode == 0 {
-		rand.Seed(time.Now().Unix())
+func GetStatusCode(statusCode string) (string, error) {
+
+	if statusCode == "" {
 		log.Info("[Info]: No status code provided. Selecting a status code randomly from supported status codes")
 		return acceptedStatusCodes[1+rand.Intn(len(acceptedStatusCodes))], nil
```
Comment (neelanjan00) [inline chaoslib/litmus/http-chaos/lib/statuscode/status-code.go:49]: If `rand.Intn(len(acceptedStatusCodes)` returns `n-1`, where `n` is the length of `acceptedStatusCodes`, then we will get an out of bound error for this expression. Can you PTAL?

File: chaoslib/litmus/http-chaos/lib/statuscode/status-code.go [hunk 2]
```diff
@@ -29,27 +35,49 @@ func PodHttpStatusCodeChaos(experimentsDetails *experimentTypes.ExperimentDetail
...
+	statusCodeList := strings.Split(statusCode, ",")
+	rand.Seed(time.Now().Unix())
+	if len(statusCodeList) == 1 {
+		if checkStatusCode(statusCodeList[0], acceptedStatusCodes) {
+			return statusCodeList[0], nil
+		}
+	} else {
+		acceptedCodes := getAcceptedCodesInList(statusCodeList, acceptedStatusCodes)
+		if len(acceptedCodes) == 0 {
+			return "", errors.New("Invalid status code provided")
```
Comment (neelanjan00) [inline chaoslib/litmus/http-chaos/lib/statuscode/status-code.go:61]:
```suggestion
			return "", errors.New("invalid status code provided")
```

File: chaoslib/litmus/http-chaos/lib/statuscode/status-code.go (hunk 1 above)
Comment (avaakash) [reply chaoslib/litmus/http-chaos/lib/statuscode/status-code.go:49]: Correct, I have updated the code. `rand.Intn(len(acceptedStatusCodes))` is what we should be using.

File: chaoslib/litmus/http-chaos/lib/statuscode/status-code.go [hunk 3]
```diff
@@ -29,27 +35,49 @@ func PodHttpStatusCodeChaos(experimentsDetails *experimentTypes.ExperimentDetail
...
+			return statusCodeList[0], nil
+		}
+	} else {
+		acceptedCodes := getAcceptedCodesInList(statusCodeList, acceptedStatusCodes)
+		if len(acceptedCodes) == 0 {
+			return "", errors.Errorf("invalid status code provided, code: %s", statusCode)
+		}
+		return acceptedCodes[rand.Intn(len(acceptedCodes))], nil
+	}
+	return "", errors.Errorf("status code %s is not supported. \nList of supported status code: %v", statusCode, acceptedStatusCodes)
```
Comment (Jonsy13) [inline chaoslib/litmus/http-chaos/lib/statuscode/status-code.go:65]:
```suggestion
	return "", errors.Errorf("status code %s is not supported. \nList of supported status codes: %v", statusCode, acceptedStatusCodes)
```

**Review by Jonsy13 — APPROVED** [review]

**Review by neelanjan00 — APPROVED** [review]

**Review by uditgaurav — APPROVED** [review]

**Pull request merged**